src/components/batlabwidget.cpp

```diff
@@ -1,30 +1,82 @@
 #include "batlabwidget.h"
 
-BatlabWidget::BatlabWidget(batlabInfo info, QWidget *parent) : QWidget(parent)
+BatlabWidget::BatlabWidget(batlabInfo info, QFrame *parent) : QFrame(parent)
 {
-    batlabWidgetLayout = new QGridLayout;
+    // **Chose to use QFrame instead of QWidget due to outline paint limitations
+    batlabLayout = new QHBoxLayout;
 
-    serialNumberLabel = new QLabel(tr("Batlab Serial Number:"));
+    // SETUP BATLAB INFO WIDGET
+    QWidget *batlabInfoWidget = new QWidget;
+
+    batlabInfoLayout = new QGridLayout;
+
+    serialNumberLabel = new QLabel(tr("Batlab Serial #:"));
     serialNumberValueLabel = new QLabel(info.serialNumberComplete == -1 ? "" : QString::number(info.serialNumberComplete));
 
-    portNameLabel = new QLabel(tr("Serial Port Name:"));
+    portNameLabel = new QLabel(tr("Port Name:"));
     portNameValueLabel = new QLabel(info.portName);
 
-    externalPowerConnectedLabel = new QLabel(tr("External Power Connected:"));
+    externalPowerConnectedLabel = new QLabel(tr("External Power:"));
     externalPowerConnectedValueLabel = new QLabel(info.externalPowerConnected ? tr("Yes") : tr("No"));
 
     firmwareVersionLabel = new QLabel(tr("Firmware Version:"));
     firmwareVersionValueLabel = new QLabel(info.firmwareVersion == -1 ? "" : QString::number(info.firmwareVersion));
+    firmwareVersionUpdateButton = new QPushButton(tr("Update Firmware"));
+    connect(firmwareVersionUpdateButton, &QAbstractButton::released, this, &BatlabWidget::updateFirmware);
+
+    // TODO: Update the current firmware version
+    if (info.firmwareVersion == 4.0) {
+        firmwareVersionUpdateButton->setEnabled(false);
```

Comment:
I could go take a closer look but it looks like this might just disable the button but leave it visible, do you think it would be better to hide it altogether (what I had originally imagined)?

Reply:
Yeah, it does just disable the button. I thought that was the goal, but I can easily just remove the button altogether. I think either way works fine, so I just removed it.